File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.java
import java.util.ArrayList;
import java.util.List;

public class Solution {

    public List<Integer> getRow(int rowIndex) {
        if (rowIndex == 0) return List.of(1);
        if (rowIndex == 1) return List.of(1, 1);

        List<Integer> prev = List.of(1, 1);
        for (int i = 2; i <= rowIndex; i++) {
            List<Integer> cur = new ArrayList<>();
            cur.add(1);
            for (int j = 1; j < i; j++) cur.add(prev.get(j - 1) + prev.get(j));
            cur.add(1);
            prev = cur;
        }

        return prev;
    }

}

File: leetcode/0119-pascals-triangle-ii.ts
function getRow(rowIndex: number): number[] {
    if (rowIndex === 0) {
        return [1];
    }

    if (rowIndex === 1) {
        return [1, 1];
    }

    let prev: number[] = [1, 1];

    for (let i = 2; i <= rowIndex; i++) {
        const cur: number[] = [];

        cur.push(1);

        for (let j = 1; j < i; j++) {
            cur.push(prev[j - 1] + prev[j]);
        }

        cur.push(1);

        prev = cur;
    }

    return prev;
}